index.js

```diff
@@ -152,12 +152,24 @@ async function init() {
 		const response = await fetch(`https://raw.githubusercontent.com/${user}/${repository}/${ref}/${escapeFilepath(file.path)}`, {
 			signal: controller.signal,
 		});
+		const responsec = response.clone();
+		const text = await response.text();
+		let blob;
+
+		if (text.startsWith('version https://git-lfs.github.com/spec/v1')) {
+			const secondaryResponse = await fetch(`https://media.githubusercontent.com/media/${user}/${repository}/${ref}/${escapeFilepath(file.path)}`, {
+				signal: controller.signal,
+			});
+			blob = await secondaryResponse.blob();
+		} else {
+			blob = await responsec.blob();
```

Comment:
I'd rather see the LFS logic extracted so that the code avoids variable names like `responsec` and a `blob` non-constant.

For example, the first fetch can just be replaced with a `fetchWithLfsSupport` function instead. Here's some untested demo code:

```js
async function fetchWithLfsSupport(url, {signal}) {
  const response = await fetch(url, {signal})
  if (isResponseMaybeLfs(response)) {
    return handleLfs(response)
  }
  
  return response;
}

function isResponseMaybeLfs(response) {
  return response.headers.contentType === 'text/plain' && response.headers.contentLength > 128 && response.headers.contentLength < 140
}

function isResponseMaybeLfs(response) {
  // your current logic, except it returns `response`
  // it must also handle the possibility that `.text()` does not start with "version etc" 
}
```

Reply:
Why override `isResponseMaybeLfs`?
The bellow one is misprint of `handleLfs`?

Reply:
Yes it's a misprint